models/issue.go

```diff
@@ -1600,28 +1604,59 @@ func SearchIssueIDsByKeyword(kw string, repoIDs []int64, limit, start int) (int6
 }
 
 // UpdateIssueByAPI updates all allowed fields of given issue.
-func UpdateIssueByAPI(issue *Issue) error {
+// If the issue status is changed a statusChangeComment is returned
+// similarly if the title is changed the titleChanged bool is set to true
+func UpdateIssueByAPI(issue *Issue, doer *User) (statusChangeComment *Comment, titleChanged bool, err error) {
```

Comment:
IMHO the creation of a comment belongs to the service module, rather than models.

...says the guy who wrote "addCrossReferences" a few lines below. 😅

Reply:
The comment is being created within the same session.